libr/core/canal.c

```diff
@@ -718,14 +713,17 @@ static int core_anal_fcn(RCore *core, ut64 at, ut64 from, int reftype, int depth
 			}
 		}
 		// TODO bring back old hack, should be fixed
-		if (!r_io_read_at (core->io, at + delta, buf, 4)) {
-			goto error;
+		{
+			ut8 buf [4];
```

Comment:
just define `buf` at the beginning of the `do/while` block, no need for another extra level of indentation just for this IMHO. It makes the code looks really ugly.

Reply:
this is just used to check if the memory can be read..in theory the is_valid_offset should be enough